pkg/state/state_rdma_device_plugin_test.go

```diff
@@ -0,0 +1,118 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state_test
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	netattdefv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	"github.com/Mellanox/network-operator/api/v1alpha1"
+	mellanoxv1alpha1 "github.com/Mellanox/network-operator/api/v1alpha1"
+	"github.com/Mellanox/network-operator/pkg/state"
+)
+
+var _ = Describe("RDMA Device Plugin rendering tests", func() {
+
+	var ctx context.Context
+	var rdmaDPState state.State
+	var rdmaDPRenderer state.ManifestRenderer
+	var testLogger logr.Logger
+	var catalog state.InfoCatalog
+	var client client.Client
+	var manifestDir string
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		scheme := runtime.NewScheme()
+		Expect(mellanoxv1alpha1.AddToScheme(scheme)).NotTo(HaveOccurred())
+		Expect(netattdefv1.AddToScheme(scheme)).NotTo(HaveOccurred())
+		client = fake.NewClientBuilder().WithScheme(scheme).Build()
+		manifestDir = "../../manifests/state-rdma-device-plugin"
+		s, r, err := state.NewStateRDMADevicePlugin(client, manifestDir)
+		Expect(err).NotTo(HaveOccurred())
+		rdmaDPState = s
+		rdmaDPRenderer = r
+		catalog = getTestCatalog()
+		testLogger = log.Log.WithName("testLog")
+	})
+
+	Context("RDMA Device Plugin state", func() {
+		It("Should Render RDMA Device Plugin Manifests", func() {
+			cr := getRDMADevicePlugin()
+			objs, err := rdmaDPRenderer.GetManifestObjects(ctx, cr, catalog, testLogger)
+			Expect(err).NotTo(HaveOccurred())
+			// We only expect a single manifest here, which should be the DaemonSet.
+			// The other manifests are only if RuntimeSpec.IsOpenshift == true.
+			Expect(len(objs)).To(Equal(1))
+			GetManifestObjectsTest(ctx, cr, catalog, &cr.Spec.RdmaSharedDevicePlugin.ImageSpec, rdmaDPRenderer)
+		})
+		It("Should Render RDMA Device Plugin Manifests for Openshift", func() {
+			cr := getRDMADevicePlugin()
+			openshiftCatalog := getOpenshiftTestCatalog()
+			objs, err := rdmaDPRenderer.GetManifestObjects(ctx, cr, openshiftCatalog, testLogger)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(objs)).To(Equal(4))
+			GetManifestObjectsTest(ctx, cr, catalog, &cr.Spec.RdmaSharedDevicePlugin.ImageSpec, rdmaDPRenderer)
+		})
```

Comment:
Don't know if we even need this test? It only verifies that there are 4 instead manifests of 1.

Reply:
since there is a difference between vanilla k8s and OpenShift deployment I would like to keep this check